src/meta/src/manager/metadata.rs

```diff
@@ -613,6 +613,22 @@ impl MetadataManager {
         }
     }
 
+    pub async fn get_running_actors_and_upstream_actors_of_fragment(
+        &self,
+        id: FragmentId,
+    ) -> MetaResult<HashSet<(ActorId, Vec<ActorId>)>> {
+        match self {
+            MetadataManager::V1(mgr) => {
+                mgr.fragment_manager
+                    .get_running_actors_and_upstream_fragment_of_fragment(id)
+                    .await
+            }
+            MetadataManager::V2(_mgr) => {
+                todo!()
```

Comment:
Can we add the implementation as well in catalog controller? We can simply query the `actor_id` and `upstream_actor_ids` columns of the `actor` table, filter by `fragment_id` and `status`.